experimental/components/src/units/converter_factory.rs

```diff
@@ -0,0 +1,339 @@
+    ///
+    /// NOTE:
+    ///   In order to have an offset, the input and output units should be simple.
+    /// This means, the input and output units should have only one unit item with power 1 and si prefix 0.
+    /// For example:
+    ///           1 - `meter` and `foot` are simple units.
+    ///           2 - `meter-per-second` and `foot-per-second` are not simple units.
+    fn compute_offset(
+        &self,
+        input_unit: &MeasureUnit,
+        output_unit: &MeasureUnit,
+    ) -> Option<Ratio<BigInt>> {
+        if !(input_unit.contained_units.len() == 1
+            && output_unit.contained_units.len() == 1
+            && input_unit.contained_units[0].power == 1
+            && output_unit.contained_units[0].power == 1
+            && input_unit.contained_units[0].si_prefix.power == 0
+            && output_unit.contained_units[0].si_prefix.power == 0)
+        {
+            return Some(Ratio::<BigInt>::from_integer(0.into()));
+        }
+
+        let input_conversion_info = self
+            .payload
+            .convert_infos
+            .get(input_unit.contained_units[0].unit_id as usize);
+        debug_assert!(
+            input_conversion_info.is_some(),
+            "Failed to get input conversion info"
+        );
+        let input_conversion_info = input_conversion_info?;
+
+        let output_conversion_info = self
+            .payload
+            .convert_infos
+            .get(output_unit.contained_units[0].unit_id as usize);
+        debug_assert!(
+            output_conversion_info.is_some(),
+            "Failed to get output conversion info"
+        );
+        let output_conversion_info = output_conversion_info?;
+
+        let input_offset = Self::extract_ratio_from_unaligned(
+            &input_conversion_info.offset_sign,
+            input_conversion_info.offset_num(),
+            input_conversion_info.offset_den(),
+        );
+
+        let output_offset = Self::extract_ratio_from_unaligned(
+            &output_conversion_info.offset_sign,
+            output_conversion_info.offset_num(),
+            output_conversion_info.offset_den(),
+        );
+
+        if input_offset.is_zero() && output_offset.is_zero() {
+            return Some(Ratio::<BigInt>::from_integer(0.into()));
+        }
+
+        let output_conversion_rate_recip = Self::extract_ratio_from_unaligned(
+            &output_conversion_info.factor_sign,
+            // Because we are computing the reciprocal, the numerator and denominator are swapped.
+            output_conversion_info.factor_den(),
+            output_conversion_info.factor_num(),
+        );
+
+        Some((input_offset - output_offset) * output_conversion_rate_recip)
+    }
+
+    // TODO(#4512): the need needs to be bikeshedded.
+    /// Checks if the given units are reciprocal or not.
+    /// If it is not reciprocal, this means that the units are convertible.
```

Comment:
If it's reciprocal, it also means that the units are convertible.

Reply:
sure, I have changed to `proportional`